opensrp-reveal/src/main/java/org/smartregister/reveal/view/ListTasksActivity.java

```diff
@@ -308,16 +319,29 @@ public void setGeoJsonSource(@NonNull FeatureCollection featureCollection, Featu
             geoJsonSource.setGeoJson(featureCollection);
             if (operationalArea != null) {
                 CameraPosition cameraPosition = mMapboxMap.getCameraForGeometry(operationalArea.geometry());
+                if (getInterventionLabel() == R.string.focus_investigation) {
+                    Feature indexCase = getIndexCase(mMapboxMap.getStyle().getSourceAs(getContext().getString(R.string.reveal_datasource_name)));
+                    if (indexCase != null) {
+                        cameraPosition = mMapboxMap.getCameraForGeometry(indexCase.geometry());
+                    }
+                } else {
+                    removeCaseIndexBoundary(mMapboxMap.getStyle());
```

Comment:
I this should be called only if one is not in a focus investigation and the the layer was already added. 
If one is in a focus investigation the GeoJSON source should be modified like the OA boundary layer

Reply:
Don't think I understand what the issue is here, could you rephrase?

Reply:
You should remove the index case layers only when a user chooses a Non FI plan. If its a FI plan then update the GeoJSON source instead of removing layers they re-adding them

Reply:
They are not removed if it is a FI. That's what the first part of the if statement looks for?